src/v/cluster/partition.cc

```diff
@@ -513,7 +513,12 @@ partition::timequery(storage::timequery_config cfg) {
         // means it _might_ hit on local data: start_timestamp is not
         // precise, so once we query we might still fall back to cloud
         // storage
-        auto result = co_await local_timequery(cfg);
+        // We also need to adjust the lower bound for the local query.
+        auto local_query_cfg = cfg;
+        local_query_cfg.min_offset = std::max(
+          log()->from_log_offset(_raft->start_offset()),
+          local_query_cfg.min_offset);
+        auto result = co_await local_timequery(local_query_cfg);
```

Comment:
Could you write a commit message for this commit? Without more context, it's easy to thinking that there are no functional changes in this commit.

Reply:
https://github.com/redpanda-data/redpanda/pull/18112/commits/f0514344c3b0672783041b782830c4b92c43681e